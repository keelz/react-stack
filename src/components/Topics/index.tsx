import * as React from 'react';

const Topics: React.SFC = () => <div className="Topics">topics</div>;

export default Topics;
