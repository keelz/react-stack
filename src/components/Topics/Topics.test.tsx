import * as React from 'react';
import * as Adapter from 'enzyme-adapter-react-16';
import { configure, shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Topics from './index';

const adapter = new Adapter();
configure({ adapter });

describe('Topics suite', () => {
  it('should match snapshot', () => {
    const tree = shallow(<Topics />);
    expect(shallowToJson(tree)).toMatchSnapshot();
  });

  it('should be selectable by .Topics', () => {
    expect(shallow(<Topics />).is('.Topics')).toBe(true);
  });

  it('should mount in a full DOM', () => {
    expect(mount(<Topics />).find('.Topics').length).toBe(1);
  });
});
