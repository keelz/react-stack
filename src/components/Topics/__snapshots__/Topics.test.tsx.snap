// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Topics suite should match snapshot 1`] = `
<div
  className="Topics"
>
  topics
</div>
`;
